### PR TITLE
ets: init at 0.2.1

### LINKED ISSUE #155074: Packaging request: ets
**Project description**

`ets` is a command output timestamper similar to `ts` in moreutils. Primary advantages over `ts` are the non-conflicting command name and its handling of progress bars. See "Comparison to moreutils ts" section in source repo for more complete comparison.

**Metadata**

* homepage URL: https://github.com/zmwangx/ets
* source URL: https://github.com/zmwangx/ets
* license: mit
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by cameronfyfe**
###### Motivation for this change

Make `ets` utility available from nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/ets/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "zmwangx";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-XWIDo5msTMTLr60x1R9cwsiZIDG6G+uHWx8idt4F2iA=";
+  };
+
+  patches = [ ./go-mod.patch ];
```
Comment (Mic92) [inline pkgs/tools/misc/ets/default.nix:14]: Can you also submit this patch upstream, unless you done so?

Comment (cameronfyfe) [reply pkgs/tools/misc/ets/default.nix:14]: Done.

File: pkgs/tools/misc/ets/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  ldflags = [ "-s" "-w" "-X main.version=v${version}-nixpkgs" ];
+
+  preBuild = ''
+    rm -rf fixtures
+  '';
+
+  postInstall = ''
+    mkdir -p $out/share/man/man1
+    cp ets.1 $out/share/man/man1
```
Comment (sikmir) [inline pkgs/tools/misc/ets/default.nix:26]:
```suggestion
    installManPage ets.1
```

File: pkgs/tools/misc/ets/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-XWIDo5msTMTLr60x1R9cwsiZIDG6G+uHWx8idt4F2iA=";
+  };
+
+  patches = [ ./go-mod.patch ];
+
+  vendorSha256 = "sha256-+8dXfqOu8XTw2uEx3GAynQSHtzifejZtddr1CdxrupA=";
+
+  ldflags = [ "-s" "-w" "-X main.version=v${version}-nixpkgs" ];
+
```
Comment (sikmir) [inline pkgs/tools/misc/ets/default.nix:19]:
```suggestion
  nativeBuildInputs = [ installShellFiles ];

```

File: pkgs/tools/misc/ets/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{ lib, buildGoModule, fetchFromGitHub }:
```
Comment (sikmir) [inline pkgs/tools/misc/ets/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```

File: pkgs/tools/misc/ets/default.nix (hunk 2 above)
Comment (cameronfyfe) [reply pkgs/tools/misc/ets/default.nix]: Thanks, I figured there was something like this but didn't find it.

File: pkgs/tools/misc/ets/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "zmwangx";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-XWIDo5msTMTLr60x1R9cwsiZIDG6G+uHWx8idt4F2iA=";
+  };
+
+  patches = [ ./go-mod.patch ];
```
Comment (Mic92) [inline pkgs/tools/misc/ets/default.nix:14]: With a PR in place this makes it easier to figure out when to drop the patch:

```suggestion
  # https://github.com/zmwangx/ets/pull/18/
  patches = [ (fetchpatch { 
    url = "https://github.com/zmwangx/ets/commit/600ec17a9c86ca63cd022d00439cdc4978e2afa9.patch";
    sha256 = "<TODO>";
  }) ];
```

File: pkgs/tools/misc/ets/go-mod.patch [hunk 6]
```diff
@@ -0,0 +1,21 @@
+diff --git a/go.mod b/go.mod
```
Comment (Mic92) [inline pkgs/tools/misc/ets/go-mod.patch:1]: This can be than also removed with `fetchpatch`.

File: pkgs/tools/misc/ets/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
```
Comment (Mic92) [inline pkgs/tools/misc/ets/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, fetchpatch, installShellFiles }:
```

File: pkgs/tools/misc/ets/default.nix (hunk 5 above)
Comment (cameronfyfe) [reply pkgs/tools/misc/ets/default.nix]: Thanks for pointing out fetchpatch

**Comment by cameronfyfe** [comment]
pulled master and rebased just now to resolve merge conflict in `all-packages.nix`

**Comment by legendofmiracles** [comment]
LGTM to me and works as expected

**Pull request merged**